lib/syscall_shim/include/uk/syscall.h

```diff
@@ -61,110 +109,79 @@ extern "C" {
-#else /* !CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS */
-/* Without userland TLS pointer support, we assume that we do not have any
- * system call that changes the TLS pointer (harmfully). This means that we do
- * not need to rely on `lib/uksched`.
- */
-#define __UK_SYSCALL_RETADDR_CLEAR();					\
-	do { _uk_syscall_return_addr = 0x0; } while (0)
-#endif /* !CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS */
-
-#else /* !CONFIG_LIBSYSCALL_SHIM */
-
-/*
- * NOTE: The usage of `uk_syscall_return_addr()` requires the `syscall_shim`
- *       library to be present. In order to avoid mis-configurations, we
- *       inject an always failing compile-time assertion to stop compilation.
- */
-#define uk_syscall_return_addr()					\
-	({ UK_CTASSERT(0x0); 0xDEADB0B0; })
-
-#define __UK_SYSCALL_RETADDR_ENTRY();					\
-	do { } while (0)
-#define __UK_SYSCALL_RETADDR_CLEAR();					\
-	do { } while (0)
-#endif /* !CONFIG_LIBSYSCALL_SHIM */
-
-#if CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS
-#include <uk/thread.h>
-
-/**
- * Library-internal thread-local variable containing the current userland
- * TLS pointer.
- * NOTE: Use the `uk_syscall_ultlsp()` macro to retrieve the userland TLS
- *       pointer.
- * NOTE: Userland TLS pointers are only supported on binary system calls.
- */
-extern __uk_tls __uptr _uk_syscall_ultlsp;
-
-/**
- * Returns the userland TLS pointer if it is not equal to the Unikraft tlsp of
- * the current thread and if it was set. Please note that we must be called
- * from a binary system call request.
- */
-static inline __uptr uk_syscall_ultlsp(void)
-{
-	__uptr ultlsp = _uk_syscall_ultlsp;
-	struct uk_thread *self = uk_thread_current();
-
-	UK_ASSERT(self);
-	if (ultlsp && (ultlsp != self->uktlsp))
-		return ultlsp;
-	return 0x0;
-}
-#else /* !CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS */
-
-/**
- * If we do not support binary system calls and userland TLS pointers, we do not
- * know any caller TLS pointer, so we can always return 0.
- */
-static inline __uptr uk_syscall_ultlsp(void)
-{
-	return 0x0;
-}
-#endif /* !CONFIG_LIBSYSCALL_SHIM_HANDLER_ULTLS */
-
 #define __uk_scc(X) ((long) (X))
 typedef long uk_syscall_arg_t;
 
 #define __UK_NAME2SCALLE_FN(name) UK_CONCAT(uk_syscall_e_, name)
 #define __UK_NAME2SCALLR_FN(name) UK_CONCAT(uk_syscall_r_, name)
 
+#define UK_USR_MAP0(...)
+/* Taken from regmap_linuxabi.h, but defined differently so as not to
+ * needlessly contaminate the namespace of source files including this header
+ */
+#if (defined __X86_64__)
+#define usr_arg0		rdi
+#define usr_arg1		rsi
+#define usr_arg2		rdx
+#define usr_arg3		r10
+#define usr_arg4		r8
+#define usr_arg5		r9
+#elif (defined __ARM_64__)
+#define usr_arg0		x[0]
+#define usr_arg1		x[1]
+#define usr_arg2		x[2]
+#define usr_arg3		x[3]
+#define usr_arg4		x[4]
+#define usr_arg5		x[5]
+#else
+#error "Missing register mappings for selected target architecture"
+#endif  /* !__X86_64__ && !__ARM_64__ */
+
+#define UK_USR_MAP2_1(m, type, arg) (long)usr->regs.usr_arg0
```

Comment:
`UK_LLSYSCALL_R_U_DEFINE` with 1 param don't work because of `UK_USR_MAP2_7` missing.

To reproduce:
```
UK_LLSYSCALL_R_U_DEFINE(int, mysyscall, unsigned long, arg)
```